### ISSUE TITLE
fix(capacitance): default `build_capacitance_solver(base_bc="fft", lambda_=0)` violates the PDE inside the mask

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/spectraldiffx#87]
Part of #80. gaussx side: jejjohnson/gaussx#260 (constant-augmented capacitance system) and jejjohnson/gaussx#259 (storage/perf).

## Summary

`build_capacitance_solver(mask, dx, dy, lambda_=0.0, base_bc="fft")` — the signature defaults — returns a field that is zero on the mask boundary (which is all the tests check) but does **not** satisfy `∇²ψ − λψ = f` in the interior. The residual is a uniform constant. The same happens with `base_bc="dct"` and `lambda_=0`. `base_bc="dst"`, or any `lambda_ > 0`, is exact.

## Reproduction

```python
import jax; jax.config.update("jax_enable_x64", True)
import jax.numpy as jnp, numpy as np
from spectraldiffx import build_capacitance_solver

ny, nx = 24, 20; dx = dy = 1.0
yy, xx = np.mgrid[:ny, :nx]
mask = ((yy - 12) ** 2 + (xx - 10) ** 2) < 7 ** 2
f = jnp.asarray(np.random.default_rng(0).standard_normal((ny, nx)))
interior = mask.copy()
interior[1:-1, 1:-1] &= mask[:-2, 1:-1] & mask[2:, 1:-1] & mask[1:-1, :-2] & mask[1:-1, 2:]
interior[[0, -1], :] = False; interior[:, [0, -1]] = False

for base in ("fft", "dct", "dst"):
    for lam in (0.0, 1.0):
        psi = np.asarray(build_capacitance_solver(mask, dx, dy, lambda_=lam, base_bc=base)(f))
        lap = (np.roll(psi, 1, 0) + np.roll(psi, -1, 0) - 2 * psi) / dy**2 \
            + (np.roll(psi, 1, 1) + np.roll(psi, -1, 1) - 2 * psi) / dx**2
        res = (lap - lam * psi - np.asarray(f))[interior]
        print(f"base={base} lam={lam}: max|res|={np.abs(res).max():.2e} mean={res.mean():+.2e}")
```

Output:

```
base=fft lam=0.0: max|res|=9.09e-04 mean=+9.09e-04    <- uniform constant
base=fft lam=1.0: max|res|=8.88e-16
base=dct lam=0.0: max|res|=8.91e-04 mean=+8.91e-04
base=dct lam=1.0: max|res|=8.88e-16
base=dst lam=0.0: max|res|=1.55e-15
base=dst lam=1.0: max|res|=6.66e-16
```

A zero-mean RHS does not fix it (residual ~3e-4). A dense masked-Dirichlet reference solve agrees with the `dst` result to 1e-15.

## Root cause

`spectraldiffx/_src/fourier/capacitance.py:184-235` builds the base solve from `solve_poisson_fft`/`solve_poisson_dct` with λ = 0. Those are pseudo-inverses that project out the constant mode, so `A B⁺ = I − 11ᵀ/n`. The capacitance correction `ψ = B⁺f − Gα` then satisfies `Aψ = f − Σα_k e_k − c·1` with `c = mean(f) − Σα_k/n`, and the `N_b`-unknown capacitance system has no degree of freedom to make `c = 0`. The generic fix (augment with the null vector and one extra unknown β) belongs in gaussx `CapacitanceSolver` (jejjohnson/gaussx#260); verified there to bring the residual to 3e-15.

## Proposed fix (spectraldiffx side)

1. Once gaussx#260 ships a `null_vector` argument: pass `jnp.ones(n)` for `base_bc="fft"` with λ = 0, and the trapezoid-weighted vector for `base_bc="dct"` (DCT-I is non-symmetric; its left null vector is `(1, 2, …, 2, 1)` per axis, outer product in 2D).
2. Until then (or as belt-and-braces): raise `ValueError` for `base_bc in ("fft", "dct")` with `lambda_ == 0`, pointing the user to `base_bc="dst"`, or change the default to `base_bc="dst"`.
3. Add the interior PDE-residual test above (all three bases × λ ∈ {0, 1}) to `tests/test_fourier_capacitance.py`; the existing tests only assert `ψ[boundary] ≈ 0`.

## Related defects in the same file (fix together)

- **`fft` base has no periodic wrap in boundary detection** (`capacitance.py:206-209`, `scipy.ndimage.binary_dilation` without wrap). Wet cells touching the rectangle edge get an unconstrained periodic neighbour. Half-plane mask 12×12, λ = 1: residual on column 0 is 2.28e-01 with `fft`, 6.7e-16 with `dst`. Either wrap-aware detection for `fft` or a validation error when a wet cell touches the edge with `base_bc="fft"`.
- **All-False mask** raises the same misleading "No inner-boundary points" error as an all-True mask; a **single-wet-cell mask** builds (N_b = 1, no interior) and silently returns zeros. Give each its own message.
- `dx`, `dy` are stored as non-static leaves but never read in `__call__` (baked into the closure), so `eqx.tree_at` on them silently does nothing; make them static or drop them.
- Docstring claims online cost O(N_b² + N log N); with the current gaussx layout it is O(N_b·N + N log N). Update after gaussx#259.

## Acceptance criteria

- [ ] Interior residual < 1e-10 for every `base_bc` × λ ∈ {0, 1} on a circular mask, an L-shaped mask, and an edge-touching mask.
- [ ] `fft` base with an edge-touching mask either wraps correctly or raises.
- [ ] Distinct errors for all-False, all-True and single-cell masks.
- [ ] gaussx pin bumped to the release containing gaussx#260.
